components/logins/src/migrate_sqlcipher_db.rs

```diff
@@ -207,14 +216,195 @@ fn generate_plan_from_db(
             }
         }
     }
+    migration_plan = apply_migration_fixups(migration_plan, &encdec)?;
     Ok(migration_plan)
 }
 
-fn migrate_logins(
-    migration_plan: &MigrationPlan,
-    store: &LoginStore,
+fn apply_migration_fixups(
+    migration_plan: MigrationPlan,
     encdec: &EncryptorDecryptor,
-) -> Result<MigrationMetrics> {
+) -> Result<MigrationPlan> {
+    let mut logins_to_override: HashMap<String, MigrationLogin> = HashMap::new();
+
+    for (guid, migration_login) in &migration_plan.logins {
+        if let (Some(local_login), Some(mirror_login)) =
+            (&migration_login.local_login, &migration_login.mirror_login)
+        {
+            // We have both a local and mirror
+            // attempt to fixup local and override mirror
+            let dec_login = local_login.login.clone().decrypt(&encdec)?;
+
+            // Skip fixup if the record is fine
+            if dec_login.entry().check_valid().is_ok() {
+                continue;
+            };
+            match dec_login.entry().fixup() {
+                Ok(new_entry) => {
+                    let login = EncryptedLogin {
+                        // record fields don't get fixed up
+                        record: RecordFields {
+                            id: dec_login.record.id,
+                            ..local_login.login.record
+                        },
+                        fields: new_entry.fields,
+                        sec_fields: new_entry.sec_fields.encrypt(encdec)?,
+                    };
+                    logins_to_override.insert(
+                        guid.to_string(),
+                        MigrationLogin {
```

Comment:
This pattern repeats a bit, but it's not clear how to avoid it - but maybe see what you can come up with - anything we can do to reduce the lines in this large function would help (although there is a balance to strike - eg, splitting it up into discrete functions would likely make everything more complex rather than less)

Reply:
For this one and the other repitition, what about switching to `migration_plan.logins.iter_mut()`?  That way I think  you could just change the `MigrationLogin` data directly.  I think it also makes it more clear which data is changing and which is staying the same.

Reply:
I'll give `iter_mut` a shot, I was wrestling with the borrow checker quite a few times to try modify the data before I gave up and went for the overriding way.